precompiles/erc20/integration_test.go

```diff
@@ -0,0 +1,1239 @@
+					fmt.Sprintf("denom is not an IBC voucher: %s", s.tokenDenom),
+				)
+				if callType == contractCall {
+					noIBCVoucherCheck = execRevertedCheck
+				}
+
+				_, _, err = s.factory.CallContractAndCheckLogs(sender.Priv, txArgs, nameArgs, noIBCVoucherCheck)
+				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
+			},
+				Entry(" - direct call", directCall),
+				Entry(" - through contract", contractCall),
+				// FIXME: Instead of "not supported" or similar this just returns the general "execution reverted" without any other info
+				// -- do we really want the same behavior for the EVM extension?
+				Entry(" - through erc20 contract", erc20Call), // NOTE: we're passing the ERC20 contract call here which was adjusted to point to a contract without metadata to expect the same errors
+			)
+
+			DescribeTable("querying the symbol should return an error", func(callType int) {
+				txArgs, symbolArgs := s.getTxAndCallArgs(callType, contractData, erc20.SymbolMethod)
+
+				noIBCVoucherCheck := failCheck.WithErrContains(
+					fmt.Sprintf("denom is not an IBC voucher: %s", s.tokenDenom),
+				)
+				if callType == contractCall {
+					noIBCVoucherCheck = execRevertedCheck
+				}
+
+				_, _, err = s.factory.CallContractAndCheckLogs(sender.Priv, txArgs, symbolArgs, noIBCVoucherCheck)
+				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
+			},
+				Entry(" - direct call", directCall),
+				Entry(" - through contract", contractCall),
+				// FIXME: Instead of "not supported" or similar this just returns the general "execution reverted" without any other info
+				// -- do we really want the same behavior for the EVM extension?
+				Entry(" - through erc20 contract", erc20Call), // NOTE: we're passing the ERC20 contract call here which was adjusted to point to a contract without metadata to expect the same errors
+			)
+
+			DescribeTable("querying the decimals should return an error", func(callType int) {
+				txArgs, decimalsArgs := s.getTxAndCallArgs(callType, contractData, erc20.DecimalsMethod)
+
+				noIBCVoucherCheck := failCheck.WithErrContains(
+					fmt.Sprintf("denom is not an IBC voucher: %s", s.tokenDenom),
+				)
+				if callType == contractCall {
+					noIBCVoucherCheck = execRevertedCheck
+				}
+
+				_, _, err = s.factory.CallContractAndCheckLogs(sender.Priv, txArgs, decimalsArgs, noIBCVoucherCheck)
+				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
+			},
+				Entry(" - direct call", directCall),
+				Entry(" - through contract", contractCall),
+				// FIXME: Instead of "not supported" or similar this just returns the general "execution reverted" without any other info
+				// -- do we really want the same behavior for the EVM extension?
+				Entry(" - through erc20 contract", erc20Call), // NOTE: we're passing the ERC20 contract call here which was adjusted to point to a contract without metadata to expect the same errors
```

Comment:
When calling the ERC20MinterBurnerDecimals contract this is just giving the generic "execution reverted" error. Do we really want to align the behaviors here and also return this non-telling error instead of a more verbose one?